libr/util/idpool.c

```diff
@@ -163,6 +162,54 @@ R_API void* r_id_storage_get(RIDStorage* storage, ut32 id) {
 	return storage->data[id];
 }
 
+R_API bool r_id_storage_get_lowest(RIDStorage *storage, ut32 *id) {
+	r_return_val_if_fail (storage, false);
+	size_t i;
+	for (i = 0; i < storage->size && !storage->data[i]; i++);
+	*id = i;
+	return i < storage->size;
+}
+
+R_API bool r_id_storage_get_highest(RIDStorage *storage, ut32 *id) {
+	r_return_val_if_fail (storage, false);
+	size_t i = 0;
+	if (storage->size > 0) {
+		for (i = storage->size - 1; !storage->data[i] && i > 0; i--);
+		*id = i;
+		return storage->data[i];  // != NULL;
+	}
+	*id = i;
+	return false;
+}
+
+R_API bool r_id_storage_get_next(RIDStorage *storage, ut32 *idref) {
```

Comment:
can we rename idref as id. Id is used as parameter in other id_storage functions. don't break that

Reply:
no. this way we avoid having to dereference a dword many unnecessary times inside the loops. the value is written back at the end.